akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorContextAskSpec.scala

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor.typed.scaladsl
+
+import akka.actor.typed.{ ActorRef, PostStop, PreRestart, Props, TypedAkkaSpec }
+import akka.pattern.AskTimeoutException
+import akka.testkit.EventFilter
+import akka.testkit.typed.TestKit
+import akka.testkit.typed.scaladsl.TestProbe
+import akka.actor.typed.scaladsl.adapter._
+import com.typesafe.config.ConfigFactory
+
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.{ Failure, Success }
+
+object ActorContextAskSpec {
+  val config = ConfigFactory.parseString(
+    """
+      akka.loggers = ["akka.testkit.TestEventListener"]
+      ping-pong-dispatcher {
+        executor = thread-pool-executor
+        type = PinnedDispatcher
+      }
+      snitch-dispatcher {
+        executor = thread-pool-executor
+        type = PinnedDispatcher
+      }
+    """)
+}
+
+class ActorContextAskSpec extends TestKit(ActorContextAskSpec.config) with TypedAkkaSpec {
+
+  implicit val untyped = system.toUntyped // FIXME no typed event filter yet
+
+  "The Scala DSL ActorContext" must {
+
+    "provide a safe ask" in {
+      case class Ping(sender: ActorRef[Pong])
+      case class Pong(selfName: String, threadName: String)
+
+      val pingPong = spawn(Behaviors.immutable[Ping] { (ctx, msg) ⇒
+        msg.sender ! Pong(ctx.self.path.name, Thread.currentThread().getName)
+        Behaviors.same
+      }, "ping-pong", Props.empty.withDispatcherFromConfig("ping-pong-dispatcher"))
+
+      val probe = TestProbe[AnyRef]()
+
+      val snitch = Behaviors.deferred[Pong] { (ctx) ⇒
+
+        // Timeout comes from TypedAkkaSpec
+
+        ctx.ask(pingPong, Ping) {
+          case Success(pong) ⇒ Pong(ctx.self.path.name + "1", Thread.currentThread().getName)
+          case Failure(ex)   ⇒ throw ex
+        }
+
+        // FIXME it is nice that it is the actorref you ask but not sure it is justified with the implicit at all WDYT?
+        import ctx.Ask
```

Comment:
I think that possibility is nice. Would it even be possible to add that implicit automatically when `ctx` is in scope?

Reply:
I'll think a bit about that.